packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -1061,7 +1061,10 @@ export class Stack extends Construct implements ITaggable {
    * Synthesizes the cloudformation template into a cloud assembly.
    * @internal
    */
-  public _synthesizeTemplate(session: ISynthesisSession, lookupRoleArn?: string): void {
+  public _synthesizeTemplate(session: ISynthesisSession,
+    lookupRoleArn?: string,
+    lookupRoleExternalId?: string,
```

Comment:
Noticed that we didn't used to pass `lookupRoleExternalId` here, which means it wasn't added to the missing context, which means that it was ignored during lookups. I reckon we didn't see this because its a pretty extreme use case and its likely no one is using it.

Reply:
This is the cause of https://github.com/aws/aws-cdk/issues/22535